python/feathub/processors/flink/table_builder/sliding_window_utils.py

```diff
@@ -105,6 +106,47 @@ def __hash__(self) -> int:
         )
 
 
+def evaluate_infinite_size_sliding_window(
+    flink_table: NativeFlinkTable,
+    window_descriptor: SlidingWindowDescriptor,
+    agg_descriptors: List[AggregationFieldDescriptor],
+) -> NativeFlinkTable:
+    """
+    Evaluate the infinite size sliding window transforms on the given flink table and
+    return the result table.
+
+    :param flink_table: The input Flink table.
+    :param window_descriptor: The descriptor of the sliding window.
+    :param agg_descriptors: A list of descriptor that descriptor the aggregation to
+                            perform.
+    :return: The result table.
+    """
+    for agg_descriptor in agg_descriptors:
+        if agg_descriptor.window_size != timedelta(seconds=0):
+            raise FeathubException(
+                "Sliding window with 0 step size must has 0 window size."
+            )
```

Comment:
Could you please briefly describe how we plan to extend this function into a public API? For example, how do we plan to support cases where window_size or step_size is 0 while the other one is not?

Reply:
According to offline discussions, we think this would for now be an internal function, for which future work is not designed as there is no need yet.